governance/program/src/state.rs

```diff
@@ -0,0 +1,239 @@
+    /// If there are outstanding signatory tokens, then cannot leave draft state. Signatories must burn tokens (ie agree
+    /// to move instruction to voting state) and bring mint to net 0 tokens outstanding. Each signatory gets 1 (serves as flag)
+    pub signatory_mint: Pubkey,
+
+    /// Admin ownership mint. One token is minted, can be used to grant admin status to a new person.
+    pub admin_mint: Pubkey,
+}
+
+/// Proposal state
+#[derive(Clone)]
+pub struct ProposalState {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Proposal account
+    pub proposal: Pubkey,
+
+    /// Current status of the proposal
+    pub status: ProposalStateStatus,
+
+    /// Total signatory tokens minted, for use comparing to supply remaining during draft period
+    pub total_signatory_tokens_minted: u64,
+
+    /// Link to proposal's description
+    pub description_link: String,
+
+    /// Proposal name
+    pub name: String,
+
+    /// When the Proposal ended voting - this will also be when the set was defeated or began executing naturally.
+    pub voting_ended_at: Option<Slot>,
+
+    /// When the Proposal began voting
+    pub voting_began_at: Option<Slot>,
+
+    /// when the Proposal entered draft state
+    pub created_at: Option<Slot>,
+
+    /// when the Proposal entered completed state, also when execution ended naturally.
+    pub completed_at: Option<Slot>,
+
+    /// when the Proposal entered deleted state
+    pub deleted_at: Option<Slot>,
+
+    /// The number of the instructions already executed
+    pub number_of_executed_instructions: u8,
+
+    /// The number of instructions included in the proposal
+    pub number_of_instructions: u8,
+
+    /// Array of pubkeys pointing at Proposal instructions, up to 5
+    pub instruction: Vec<Pubkey>,
+}
+
+/// What state a Proposal is in
+#[derive(Clone, Debug, PartialEq)]
+pub enum ProposalStateStatus {
+    /// Draft - Proposal enters Draft state when it's created
+    Draft,
+
+    /// Signing - The Proposal is being signed by Signatories. Proposal enters the state when first Signatory Sings and leaves it when last Signatory signs
+    Signing,
+
+    /// Taking votes
+    Voting,
+
+    /// Voting ended with success
+    Succeeded,
+
+    /// Voting completed and now instructions are being execute. Proposal enter this state when first instruction is executed and leaves when the last instruction is executed
+    Executing,
+
+    /// Completed
+    Completed,
+
+    /// Cancelled
+    Cancelled,
+
+    /// Defeated
+    Defeated,
+}
+
+impl Default for ProposalStateStatus {
+    fn default() -> Self {
+        ProposalStateStatus::Draft
+    }
+}
+
+/// Governance Vote Record
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct ProposalVoteRecord {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Proposal account
+    pub proposal: Pubkey,
+
+    /// The user who casted this vote
+    pub voter: Pubkey,
```

Comment:
What address is this exactly?  `voter` is really generic, is this the `VoterRecord` account address?  If so, call this field `voter_record`

Reply:
This is the owner of the deposited governing token. I renamed it to `governing_token_owner` and updated comments to make it clearer.